google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BigtableStackdriverExportUtils.java

```diff
@@ -0,0 +1,312 @@
+import com.google.common.collect.Maps;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import io.opencensus.common.Functions;
+import io.opencensus.metrics.LabelKey;
+import io.opencensus.metrics.LabelValue;
+import io.opencensus.metrics.data.AttachmentValue;
+import io.opencensus.metrics.export.Distribution.Bucket;
+import io.opencensus.metrics.export.Distribution.BucketOptions.ExplicitOptions;
+import io.opencensus.metrics.export.MetricDescriptor.Type;
+import io.opencensus.metrics.export.Summary;
+import io.opencensus.metrics.export.Value;
+import java.lang.management.ManagementFactory;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+class BigtableStackdriverExportUtils {
+  private static final Logger logger;
+
+  private static final io.opencensus.common.Function<Double, TypedValue> typedValueDoubleFunction;
+  private static final io.opencensus.common.Function<Long, TypedValue> typedValueLongFunction;
+  private static final io.opencensus.common.Function<
+          io.opencensus.metrics.export.Distribution, TypedValue>
+      typedValueDistributionFunction;
+  private static final io.opencensus.common.Function<Summary, TypedValue> typedValueSummaryFunction;
+  private static final io.opencensus.common.Function<ExplicitOptions, BucketOptions>
+      bucketOptionsExplicitFunction;
+
+  private static final Set<String> PROMOTED_RESOURCE_LABELS =
+      ImmutableSet.of(
+          BuiltinMeasureConstants.PROJECT_ID.getName(),
+          BuiltinMeasureConstants.INSTANCE_ID.getName(),
+          BuiltinMeasureConstants.CLUSTER.getName(),
+          BuiltinMeasureConstants.ZONE.getName(),
+          BuiltinMeasureConstants.TABLE.getName());
+
+  static String getDefaultTaskValue() {
+    String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+    if (jvmName.indexOf(64) < 1) {
+      String hostname = "localhost";
+
+      try {
+        hostname = InetAddress.getLocalHost().getHostName();
+      } catch (UnknownHostException e) {
+        logger.log(Level.INFO, "Unable to get the hostname.", e);
+      }
+
+      return "java-" + (new SecureRandom()).nextInt() + "@" + hostname;
+    } else {
+      return "java-" + jvmName;
+    }
+  }
+
+  @VisibleForTesting
+  static MetricKind createMetricKind(Type type) {
+    switch (type) {
+      case GAUGE_INT64:
+      case GAUGE_DOUBLE:
+        return MetricKind.GAUGE;
+      case CUMULATIVE_DOUBLE:
+      case CUMULATIVE_INT64:
+      case CUMULATIVE_DISTRIBUTION:
+        return MetricKind.CUMULATIVE;
+      default:
+        return MetricKind.UNRECOGNIZED;
```

Comment:
I think you only care about cumulative

Reply:
Do we want to keep both in case we add new metrics in the future?